### ISSUE TITLE
Extra weak (frail/anemic/???) tags, or improved declaration syntax.

### ISSUE THREAD

**Issue by Y-Less** [pawn-lang/compiler#383]
### Issue description:

Any tag that starts with an upper-case letter is strong.  ALL operations involving mixing these are warnings:

```pawn
new Text:a = 5; // Warning.
new b = Text:6; // Warning.
```

Any tag that starts with a lower-case letter is weak.  SOME operations involving mixing these are warnings:

```pawn
new bool:c = 5; // Warning.
new d = true; // NO Warning.
```

Going from a weak tag to `_` (no tag) is generally not a warning, but from `_` to a weak tag is.  One obtuse use for tags is in complex macros, since they can be chained.  But these almost always end with `_:` to remove all the tags again.  However, some times this is not possible, for example when declaring a variable.  This is a syntax error:

```pawn
new _:OTHER_TAG:e;
```

It would be very nice to be able to use more tag macros on variable declarations (and other places) without having to worry about tag mismatches at all.  I see two possible solutions currently (there may be more):

1. The `_` prefix doesn't really fit in to either "strong" or "weak" categories (they are currently classified as weak, but seem like an outlier), so I propose a third category - "impotent" tags.  They are not `_` but give no warnings when mixed with `_` ever.  These would be any tag starting with `_`:

```
new _MACRO:f = 55; // NO Warning.
new g = _MACRO:55; // NO Warning.
```

Pro: Allows macros to manipulate the true tag of the variable.
Con: Possibly breaking change for any existing tags with `_` prefix (but `_` prefixes are supposed to be reserved anyway, so this would just be assigning language semantics to previously reserved syntax).

2. Just allow multiple tags on variable declarations, with the last one always being used as the true type, as in expressions:

```
new _:MACRO:h = 55; // NO Warning.
```

Pro: Less of a breaking change, it is relaxing rules.
Con: Code such as in the example would be fixed to a tag of `_:`, since the final tag appears before any macro tags.

### Minimal complete verifiable example (MCVE):

```
new _MACRO:i = 55;
```

Or:

```
new _:MACRO:j = 55;
```

### Workspace Information:

* Compiler version:
* Command line arguments provided (or sampctl version):
* Operating System:

**Comment by Y-Less** [comment]
Or even better - tags that are SO weak they do absolutely nothing at all:

```
new _tag_1:_tag_2:Group:v; // Tag of `v` is `Group`.
```

They don't modify the current tag in any way what so ever.  Or maybe use the `@` tag prefix for that (since it is the other outlier from the strong/weak upper/lower rule).

**Comment by Southclaws** [comment]
I like the idea of silent/quiet tags. Multiple tags sounds interesting but the most complicated to implement. The use of prefixes seems to work fine with the existing semantics of strong/week being controlled by the first character (it's kinda similar to how Go controls exposure!)
